Userland/Services/FileSystemAccessServer/DownloadWindow.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2021, timmot <tiwwot@protonmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "DownloadWindow.h"
+#include "DownloadWidget.h"
+#include <AK/HashMap.h>
+
+namespace FileSystemAccessServer {
+
+static HashMap<u32, RefPtr<DownloadWindow>> s_windows;
+
+DownloadWindow::DownloadWindow(i32 client_id, URL const& url, String const& destination_path)
+{
+    m_id = client_id;
+    s_windows.set(m_id, this);
```

Comment:
```suggestion
    s_windows.set(m_id, *this);
```
This is needed because a `NonnullRefPtr` cannot be constructed from a raw pointer, but it can be from a reference (since it's guaranteed non-null).